kratos/includes/model_part.h

```diff
@@ -568,6 +599,111 @@ class KRATOS_API(KRATOS_CORE) ModelPart : public DataValueContainer, public Flag
     }
 
 
+    ///@}
+    ///@name MasterSlaveConstraints
+    ///@{
+
+    SizeType NumberOfMasterSlaveConstraints() const
+    {
+        return mMasterSlaveConstraints.size();
+    }
+
+    MasterSlaveConstraintContainerType& MasterSlaveConstraints()
+    {
+        return mMasterSlaveConstraints;
+    }
+
+    MasterSlaveConstraintContainerType::Pointer pGetMasterSlaveConstraints()
+    {
+        return Kratos::shared_ptr<MasterSlaveConstraintContainerType>(&mMasterSlaveConstraints);
+    }
+
+    void SetMasterSlaveConstraints(MasterSlaveConstraintContainerType::Pointer pOtherMasterSlaveConstraints)
+    {
+        mMasterSlaveConstraints = *pOtherMasterSlaveConstraints;
```

Comment:
Also here I think you should pass by reference in the function parameter, not by shared pointer since you are not passing ownership.